### ISSUE TITLE
Markdown vs ReStructuredText

### ISSUE THREAD

**Issue by noughtmare** [ghc-proposals/ghc-proposals#560]
Continuing the discussion from https://github.com/ghc-proposals/ghc-proposals/pull/555#issuecomment-1332594529 into its own ticket.

I made the choice to use markdown after reading this in the readme:

> Proposals are written in either [ReStructuredText](http://www.sphinx-doc.org/en/stable/rest.html) or [Markdown](https://github.github.com/gfm/). While the proposal process itself has no preference, keep in mind that the [GHC Users Guide](http://downloads.haskell.org/~ghc/latest/docs/html/users_guide/editing-guide.html) uses ReStructuredText exclusively. Accepted proposals written in ReStructuredText thus have the slight benefit that they can be more easily included in the official GHC documentation.

This gave me the impression that the only advantage of RST is that it can more easily be moved into the users guide. And I knew I didn't need to do that for #555, so I chose markdown.

Now it seems there is another big advantage of RST: it has numbered sections and an automatic table of contents.

I would propose changing the readme to mention that RST is preferred for that reason. I don't think markdown should be completely disallowed, because that might form another obstacle for some people. But at least the proposal process could have a clear preference.

**Comment by adamgundry** [comment]
It turns out GitHub does generate a table of contents for Markdown files (and RST files), it is just cleverly hidden: https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/

I agree we should change the wording to state that RST is preferred, but that Markdown is acceptable. And perhaps we should encourage authors of Markdown proposals to number their sections manually?

**Comment by nomeata** [comment]
> I would propose changing the readme to mention that RST is preferred for that reason. I don't think markdown should be completely disallowed, because that might form another obstacle for some people. But at least the proposal process could have a clear preference.

Agreed. Do you want to create a PR?

**Comment by Ericson2314** [comment]
Interesting find from @adamgundry, but I think it is nice to see without clicking a button and also the original motivation of making stuff easy to move to the users guide tips my scales. Manual numbering is laborious to maintain during edits, so markdown authors would probably regret their choice anyways, and clutters diffs which is bad for review.

**Comment by AntC2** [comment]
Originally when the more formal review/SC voting process was formalised, proposals had to be in ReST only. I never quite understood why relax to allow Markdown as well. Was ReST seen as too much of a barrier? But don't we _want_ a <s>barrier</s>high bar to make sure the proposer is diligent? (Typing up/formatting the proposal is relatively easy work compared to thinking through all the 'effects and interactions'.)

**Comment by nomeata** [comment]
@isovector proposed it in https://github.com/ghc-proposals/ghc-proposals/pull/250. I'm not attached to offering the markdown variant, if it causes more friction than benefit I'm happy to remove it.

**Comment by goldfirere** [comment]
There is a difficult balance between ease of proposing and rigor. In the past, I've been an advocate for lowering the barriers to proposals as much as possible. I continue to believe that barriers are bad, but I now see that some of the challenge is inherent in the process. That is, language design is a difficult craft, and expertise matters. Presenting the situation differently just exposes contributors to disappointment at writing a proposal and then struggling to find a proper reception for it.

Point of contrast: I understand that JavaScript requires a full implementation of a feature before considering a proposal. I'm not at all suggesting that here, but just making a point that requiring a particular format and some BNF is not really all that high a bar.

Bottom line for me: I'd support going back to just ReST. After all, the markup language is by far not the dominant obstacle here!

**Comment by nomeata** [comment]
Ok, let's do that. We can still revert if people complain.

**Comment by VitWW** [comment]
For whom is interested in it: see #712

**Comment by adamgundry** [comment]
Closing as in #712 we removed the Markdown template in favour of using RST uniformly.
